packages/mobile/src/account/Settings.test.tsx

```diff
@@ -90,4 +99,33 @@ describe('Account', () => {
     )
     expect(tree).toMatchSnapshot()
   })
+
+  it('navigates to PincodeSet screen if entered PIN is correct', async () => {
+    const tree = render(
+      <Provider store={createMockStore({})}>
+        <Settings {...getMockStackScreenProps(Screens.Settings)} />
+      </Provider>
+    )
+    mockedNavigate.mockImplementationOnce(() => {
+      Promise.resolve(true)
+    })
+    fireEvent.press(tree.getByTestId('ChangePIN'))
+    expect(navigate).toHaveBeenCalledWith(Screens.PincodeSet, {
+      isVerifying: false,
+      changePin: true,
+    })
+  })
+
+  it('does not navigate to PincodeSet screen if entered PIN is incorrect', async () => {
+    const tree = render(
+      <Provider store={createMockStore({})}>
+        <Settings {...getMockStackScreenProps(Screens.Settings)} />
+      </Provider>
+    )
+    mockedNavigate.mockImplementationOnce((error) => {
+      return error
```

Comment:
The `navigate` doesn't receive an `error` param, we should mock `ensurePincode` instead of `navigate` and have this throw an error or return `Promise.reject()`

Reply:
ah i see, does the updated code make sense?